File: Elixr.Api/Models/Armor.cs
using System;

namespace Elixr.Api.Models
{
    public class Armor : Equipment, IVotable
    {
        public int DefenseBonus {get;set;}
        public int SpeedPenalty { get; set;}
        public int RoundsToDon
        {
            get
            {
                if (WeightInPounds < 25)
                {
                    return 5;
                }
                if (WeightInPounds < 45)
                {
                    return 10;
                }
                return 20;
            }
        }
        public bool BestowsDisadvantage 
        {
            get
            {
                return this.WeightInPounds >= 45;
            }
        }
    }
}

File: Elixr.Website/models/armor.ts
export default class Armor implements Elixr.Api.ViewModels.ArmorViewModel {

    author: Elixr.Api.ViewModels.PlayerViewModel;
    cost = 0;
    createdAtMS = 0;
    description = "";
    enchantments: Elixr.Api.ViewModels.EnchantmentViewModel[];
    equipmentId = 0;
    imageUrl = "";
    name = "";
    weightInPounds = 0;
    defenseBonus = 0;
    speedPenalty = 0;

    get roundsToDon(): number {
        if (this.weightInPounds < 25) {
            return 5;
        }
        if (this.weightInPounds < 45) {
            return 10;
        }
        return 20;
    }
    get bestowsDisadvantage(): boolean {
        return this.weightInPounds >= 45;
    }

}